### PR TITLE
Add Instructions for Model Retraining

### PR DESCRIPTION
**Pull request by anniestaker**

### CODE REVIEW

File: examples/Model_Retrain_Instructions.pdf
Comment (sarah-innis) [inline examples/Model_Retrain_Instructions.pdf:1]: instead of referring to the jupyter notebook for the training my thought would be to refer them to these functions instead? [walandmarks](https://github.com/Ivaldivi/DATA515-Image-Classification/tree/annie/retrain-example/walandmarks)/[helpers](https://github.com/Ivaldivi/DATA515-Image-Classification/tree/annie/retrain-example/walandmarks/helpers)
/landmark_classification_model_training.py I think you would just run lines 304 to 342?

Comment (Ivaldivi) [reply examples/Model_Retrain_Instructions.pdf:1]: Yes, in theory, if someone wants to retrain the model they would just call create_train_analyze_model from that .py file. I can add a main guard (if __name__ == '__main__') that contains the function call so it's ready for the model-maintainer but doesn't get called during unittesting

Comment (anniestaker) [reply examples/Model_Retrain_Instructions.pdf:1]: Okay. I added a main guard and did some tweaking of the function. I also updated the doc strings and tests.

Then I updated the instruction file to use this function instead of the notebook.

File: walandmarks/test/test_model_training.py
Comment (Ivaldivi) [inline walandmarks/test/test_model_training.py:1]: did you remove the test that checks if dropout rate is a float on purpose?

File: walandmarks/test/test_model_training.py [hunk 1]
```diff
@@ -298,17 +300,31 @@ def test_create_train_analyze_model_raises_type_error_if_save_model_pathname_not
         if path to save model is not a string. 
         """
         with self.assertRaises(TypeError):
-            cm.create_train_analyze_model(False, 5)
+            cm.create_train_analyze_model(False, 5, 'loss/path', 'accuracy/path', 'auc/path')
 
-    def test_create_train_analyze_model_raises_type_error_if_dropout_rate_not_float(self):
```
Comment (Ivaldivi) [inline walandmarks/test/test_model_training.py:303]: did you remove the test that checks if dropout rate is a float on purpose?

Comment (anniestaker) [reply walandmarks/test/test_model_training.py:303]: Yes I did. I removed it as a parameter from the function and just passed the parameter when it was needed. Because of that, the unit test was no longer applicable. I did this because pylint wouldn't let me have more than five arguments in a function and I added paths to save all the plots that were being generated.

Honestly, I think a better thing to do than this would be to create a configuration file with all those parameters (save paths, input paths, model hyperparameters, etc.) and load the configuration file instead of passing parameters individually into the function. Would it be better if I implemented that approach?

Comment (Ivaldivi) [reply walandmarks/test/test_model_training.py:303]: Ah, I see. Thanks for the clarification. I think it's fine as is. We chatted offline and I'm going to approve and merge.

**Review by Ivaldivi — APPROVED** [review]
